packages/qdrant-loader/src/qdrant_loader/core/worker/queue.py

```
@@ -147,82 +152,85 @@ async def claim_next(
-            result = await session.execute(
-                update(Job)
-                .where(*where_clauses)
-                .values(
-                    status=self.RUNNING,
-                    started_at=now,
-                    finished_at=None,
-                    visibility_deadline=visibility_deadline,
-                    attempts=Job.attempts + 1,
-                    last_error=None,
+        async with self._db_op_lock:
+            async with self._session_factory() as session:
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)